python/cudf/cudf/core/multiindex.py

```diff
@@ -772,7 +773,20 @@ def _compute_validity_mask(self, index, row_tuple, max_length):
             ],
             axis=1,
         )
-        result = lookup.merge(data_table)["idx"]
+        # Sort indices in pandas compatible mode
+        # because we want the indices to be fetched
+        # in a deterministic order.
+        # TODO: Remove this after merge/join
+        # obtain deterministic ordering.
+        if cudf.get_option("mode.pandas_compatible"):
+            lookup_order = "_" + "_".join(map(str, lookup.columns))
```

Comment:
We will need to use `lookup._data.names` here instead of `lookup.columns`

Reply:
Ah, thanks.